src/include/sof/lib/ams.h

```diff
@@ -0,0 +1,340 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright (c) 2022 Intel Corporation
+ *
+ * Author: Krzysztof Frydryk <krzysztofx.frydryk@intel.com>
+ *
+ */
+
+#ifndef AM_SERVICE_H
+#define AM_SERVICE_H
+
+#include <errno.h>
+#include <sof/schedule/task.h>
+#include <ipc/topology.h>
+#include <rtos/alloc.h>
+#include <sof/coherent.h>
+#include <sof/lib/uuid.h>
+
+#define AMS_INVALID_MSG_TYPE 0
+#define AMS_SERVICE_UUID_TABLE_SIZE 49
+#define AMS_INVALID_SLOT 0xFF
+#define AMS_ANY 0xFFFF
+
+#define AMS_ROUTING_TABLE_SIZE 41
+#define AMS_MAX_MSG_SIZE 0x1000
+
+#define AMS_SLOT_SIZE(msg) (AMS_MESSAGE_SIZE(msg) + sizeof(uint16_t) * 2)
```

Comment:
sizeof(uint32_t)?

What that the x2 mean here?

Reply:
add a comment and explain why we've not using AMS_MAX_MSG_SIZE directly.

Reply:
Ok, though this is pretty self explanatory.